hail/Makefile

```diff
@@ -268,13 +268,13 @@ $(eval $(call ENV_VAR,cloud_base))
 $(eval $(call ENV_VAR,wheel_cloud_path))
 
 python/hailtop/hailctl/deploy.yaml: env/cloud_base env/wheel_cloud_path
-python/hailtop/hailctl/deploy.yaml: $(resources) python/requirements.txt
+python/hailtop/hailctl/deploy.yaml: $(resources) python/pinned-requirements.txt
```

Comment:
this target does not exist.

Reply:
This seems like a foot-gun to me. Suppose I'm trying to experiment with a modified version of Hail. I've modified `python/requirements.txt` but I've not run `generate-pip-lockfile`. I run `make install-hailctl` thinking that this will incorporate my changes into a new wheel file which I can use to start cluster. In reality, my changes are ignored and the old pinned requirements are used.

I think we need a rule like:

```
python/pinned-requirements.txt: python/requirements.txt
	docker run --rm -it \
		-v $(HAIL_HAIL_DIR):/hail \
		python:3.7-slim \
		/bin/bash -c "pip install pip-tools && cd /hail && \
	pip-compile --upgrade python/requirements.txt \
		--output-file=python/pinned-requirements.txt"
```

Or a rule like:

```
python/pinned-requirements.txt: generate-pip-lockfile
```

This latter rule will trigger `generate-pip-lockfile` to always be run if we depend on the pinned-requirements.txt which isn't great. I think the best option is to have a rule for each requirements file and to also have generate-pip-lockfile as a special fast path for developers who want to update all the lock files.

Reply:
Thanks for the clarification, this makes a lot of sense! I will incorporate these changes.